test/unit/controllers/declaration/ConsignmentReferencesControllerSpec.scala

```diff
@@ -63,22 +64,56 @@ class ConsignmentReferencesControllerSpec extends ControllerSpec {
       }
     }
 
-    "return 400 (BAD_REQUEST)" in new SetUp {
+    "return 400 (BAD_REQUEST)" when {
+      "no UCR is submitted" in new SetUp {
 
-      val incorrectForm = Json.toJson(ConsignmentReferences(Some(Ducr("1234")), ""))
+        val incorrectForm = Json.toJson(ConsignmentReferences(Ducr("1234"), ""))
 
-      val result = controller.submitConsignmentReferences()(postRequest(incorrectForm))
+        val result = controller.submitConsignmentReferences()(postRequest(incorrectForm))
 
-      status(result) must be(BAD_REQUEST)
+        status(result) must be(BAD_REQUEST)
+      }
+
+      "UCR length has been exceeded" in new SetUp {
+
+        val incorrectForm =
+          Json.toJson(ConsignmentReferences(Ducr("1234"), "", Some(TestHelper.createRandomAlphanumericString(36))))
+
+        val result = controller.submitConsignmentReferences()(postRequest(incorrectForm))
+
+        status(result) must be(BAD_REQUEST)
+      }
+
+      "UCR has special characters" in new SetUp {
+
+        val incorrectForm =
+          Json.toJson(ConsignmentReferences(Ducr("1234"), "", Some("special@&%")))
+
+        val result = controller.submitConsignmentReferences()(postRequest(incorrectForm))
+
+        status(result) must be(BAD_REQUEST)
+      }
```

Comment:
Those tests still exists. I still think it's unnecessary, because every test cover the same part of controller codebase and it should be in form spec :)